packages/node_modules/@webex/widget-space/src/container.js

```diff
@@ -143,19 +144,55 @@ export class SpaceWidget extends Component {
       const callStartTime = (call?.activeParticipantsCount > 0) ? call?.startTime : null;
 
       let menuButton;
+      const {preferredWebexSite} = sparkInstance.meetings;
+      let isMeetButtonDisabled = false;
+      const meetings = sparkInstance.meetings.getAllMeetings();
+
+      if (Object.keys(meetings).length === 0) {
+        isMeetButtonDisabled = !preferredWebexSite;
+      }
```

Comment:
i am a little confused with this logic shouldn't it be 

const meetings = sparkInstance.meetings.getAllMeetings();

if((sparkInstance.meetings.getMeetingByType(CONVERSATION_URL, conversation URL of the space you in) {
   show the meet button
}
else  {
if (have meeting preference ){
show meet button 
}
else {
don't show meet 
}
}

Reply:
@arun3528,
Code needs little modification, But I am also doing the same,

if (Object.keys(meetings).length === 0) {
        isMeetButtonDisabled = !preferredWebexSite;
      }
This checks if any meeting is present, if not check if preferredWebexSite is available or not if not available disable the meet button,
In the else part, We check if a meetings is available,  loop through all check the conv URL and check getMeetingByType if this returns show the meet button don't disable.

for (const key of Object.keys(meetings)) {
          if (sparkInstance.meetings.getMeetingByType(CONVERSATION_URL, key)) {
            isMeetButtonDisabled = false;
            break;
          }
        }

Reply:
looks good , lets test it out

Reply:
do you know why test is failing ?